device/sniff.go

```diff
@@ -0,0 +1,89 @@
+package device
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/Vworri/GoEavesDrop/packet"
+)
+
+type SniffProcess struct {
+	//process information for sniff commands
+	name          string
+	Process       *os.Process
+	StartTime     time.Time
+	EndTime       time.Time
+	FilePath      string
+	Duration      int
+	ContentType   []string
+	Command       *exec.Cmd
+	IsPromiscuous bool
+	Save          bool
+	SaveLocation  string
+	OutputStream  io.ReadCloser
+	ErrorStream   io.ReadCloser
+	Queue         []*packet.Packet
+}
+
+func (device *Dev) Sniff() {
+	var sniff SniffProcess
+	sniff.Command = exec.Command("sudo", "tshark", "-l", "-p", "-V", "-a", "duration:20", "-S", "\"===== THIS IS THE END ===\"")
+
+	// Deal with output stream
+	var err error
+	sniff.OutputStream, err = sniff.Command.StdoutPipe()
+	if err != nil {
+		log.Fatal("Cannot pipe standard output of sniff")
```

Comment:
Do you want to do anything to indicate why not? The err might have good clues you could combine with this message in an `fmt.Errorf()`

Reply:
I ain't fancy, I'll just return the  error